docs/sdk/src/guides/parachains/async_backing_guide.rs

```diff
@@ -0,0 +1,368 @@
+//!   instance of `FixedVelocityConsensusHook` is defined with both a fixed capacity and velocity. It
+//!   aborts the runtime early if either capacity or velocity is exceeded, as the collator shouldn’t be
+//!   creating additional blocks in that case.
+//! - **AsyncBackingParams.max_candidate_depth** - This parameter determines the maximum unincluded
+//!   segment depth the relay chain will support. Candidates sent to validators which exceed
+//!   `max_candidate_depth` will be ignored. `Capacity`, as mentioned above, should not exceed
+//!   `max_candidate_depth`.
+//! - **AsyncBackingParams.allowed_ancestry_len** - Each parachain block candidate has a `relay_parent`
+//!   from which its execution and validation context is derived. Before async backing the
+//!   `relay_parent` for a candidate not yet backed was required to be the fresh head of a fork. With
+//!   async backing we can relax this requirement. Instead we set a conservative maximum age in blocks
+//!   for the `relay_parent`s of candidates in the unincluded segment. This age, `allowed_ancestry_len`
+//!   lives on the relay chain and is queried by parachains when deciding which block to build on top
+//!   of.
+//! - **Lookahead Collator** - A collator for Aura that looks ahead of the most recently included
+//!   parachain block when determining what to build upon. This collator also builds additional blocks
+//!   when the maximum backlog is not saturated. The size of the backlog is determined by invoking the
+//!   AuraUnincludedSegmentApi. If that runtime API is not supported, this assumes a maximum backlog
+//!   size of 1.
+//!
+//! ## Prerequisite
+//!
+//! The relay chain needs to have async backing enabled so double-check that the relay-chain
+//! configuration contains the following three parameters (especially when testing locally e.g. with
+//! zombienet):
+//!
+//! ```json
+//! "async_backing_params": {
+//!     "max_candidate_depth": 3,
+//!     "allowed_ancestry_len": 2
+//! },
+//! "scheduling_lookahead": 2
+//! ```
+//!
+//! <div class="warning">`scheduling_lookahead` must be set to 2, otherwise parachain block times will degrade to worse than with sync backing!</div>
+//! 
+//!  
+//! 
+//! 
+//!
+//! ## Phase 1 - Update Parachain Runtime
+//!
+//! This phase involves configuring your parachain’s runtime `/runtime/src/lib.rs` to make use of async backing system.
+//!
+//! 1. Establish and ensure constants for `capacity` and `velocity` are both set to 1 in
+//!    `/runtime/src/lib.rs`.
+//!    
+//!    ```text
+//!    //!/ Maximum number of blocks simultaneously accepted by the Runtime, not yet included into the
+//!    //!/ relay chain.
+//!    pub const UNINCLUDED_SEGMENT_CAPACITY: u32 = 1;
+//!    //!/ How many parachain blocks are processed by the relay chain per parent. Limits the number of
+//!    //!/ blocks authored per slot.
+//!    ```
+#![doc = docify::embed!("../../templates/parachain/runtime/src/lib.rs", velocity)]
```

Comment:
<img width="1029" alt="Screenshot 2024-05-09 at 12 52 20" src="https://github.com/paritytech/polkadot-sdk/assets/49718502/8fbf1ab1-5b2d-46d7-83bf-9b9608a951a9">

This rendering is weird and there are a few other places where this happen.

I don't think we get any benefit from referencing constants in the templates parachains, this guide is for parachains that don't use the template.

Reply:
You're right. I tried to use docify where I could, and like I noted here https://github.com/paritytech/polkadot-sdk/pull/4363#issuecomment-2091891735 it may not be necessary. Let me work on a better rendering here and then decide if we can merge this or just keep the guide on the Wiki.